RCTAppleHealthKit/RCTAppleHealthKit+Methods_Dietary.m

```diff
@@ -406,4 +416,320 @@ - (void)saveWater:(NSDictionary *)input callback:(RCTResponseSenderBlock)callbac
+    
+    HKCorrelationType *foodtype = [HKCorrelationType correlationTypeForIdentifier:HKCorrelationTypeIdentifierFood];
+    NSLog(@"%@", foodtype);
+    
+    NSUInteger limit = [RCTAppleHealthKit uintFromOptions:input key:@"limit" withDefault:HKObjectQueryNoLimit];
+    BOOL ascending = [RCTAppleHealthKit boolFromOptions:input key:@"ascending" withDefault:false];
+    NSDate *startDate = [RCTAppleHealthKit dateFromOptions:input key:@"startDate" withDefault:nil];
+    NSDate *endDate = [RCTAppleHealthKit dateFromOptions:input key:@"endDate" withDefault:[NSDate date]];
+
+    NSLog(@"%@", startDate);
+    NSLog(@"%@", endDate);
+    NSPredicate *predicate = [HKQuery predicateForSamplesWithStartDate:startDate endDate:endDate options:HKQueryOptionNone];
+    //NSLog(@"%@", endDate);
+    
+    if(startDate == nil) {
+        callback(@[RCTMakeError(@"StartDate is required in options", nil, nil)]);
+        return;
+    }
+    
+//    NSPredicate *predicate = [HKQuery predicateForSamplesWithStartDate:startDate endDate:endDate options:HKQueryOptionNone];
+//    NSPredicate *userEnterDataPredicate = [HKQuery predicateForObjectsWithMetadataKey:HKMetadataKeyWasUserEntered operatorType:NSEqualToPredicateOperatorType value: [NSNumber numberWithBool:true]];
+//    NSPredicate *allPredicates = [NSCompoundPredicate andPredicateWithSubpredicates:[NSArray arrayWithObjects:predicate,userEnterDataPredicate, nil]];
+    
+    [self fetchCorrelationFoodSamplesOfType:foodtype predicate:predicate ascending:ascending limit:limit completion:^(NSArray *results, NSError *error) {
+        if (results) {
+            NSLog(@"%@", results);
+
+            if (results == nil) {
+                // Provide proper error handling here...
+                NSLog(@"An error occurred while searching for blood pressure data %@",
+                      error.localizedDescription);
+                abort();
+            }
+            
+            NSMutableDictionary *aFoodData = [[NSMutableDictionary alloc] init];
+            NSMutableArray *aNutration = [[NSMutableArray alloc] init];
+            
+            for (int i =0; i<results.count ; i++) {
+                HKCorrelation *correlation = results[i][@"correlation"];
+
+                NSSet *objs = correlation.objects;
+                for (HKQuantitySample *sample in objs) {
+                    
+                    NSLog(@"%@", sample);
+                    
+                    NSString *foodType = sample.metadata[HKMetadataKeyFoodType];
+                    NSString *foodMeal = sample.metadata[@"HKFoodMeal"];
+                    NSString *foodDate = [NSString stringWithFormat:@"%@", sample.startDate];
+                    
+                    double quantity = 0.0;
+                    NSString *type = @"";
+                   
+                    if ([[NSString stringWithFormat:@"%@", sample.quantityType] isEqualToString: HKQuantityTypeIdentifierDietaryBiotin]) {
```

Comment:
Can we use the switch case instead of if-else here?
Also, most the code is within the if conditions have been used duplicate (can we can create a function and pass the `type` value as a parameter to that function)